src/pkg/packager/lint/lint.go

```diff
@@ -50,6 +53,61 @@ func ValidateZarfSchema(path string) (*Validator, error) {
 	return &validator, nil
 }
 
+func isPinnedImage(image string) bool {
+	if strings.HasSuffix(image, ":latest") {
+		return false
+	}
+	// Pinned with image digest
+	if strings.Contains(image, "@") {
+		return true
+	}
+	splitImage := strings.Split(image, ":")
+	if len(splitImage) > 1 {
+		// In this case we either have a port and a "/" afterwards for the path to the image
+		// Or we have a tag which cannot have the / character in it
+		if !strings.Contains(splitImage[len(splitImage)-1], "/") {
+			return true
+		}
+	}
+	return false
+}
+
+func isPinnedRepo(repo string) bool {
+	// Pinned github and dev.azure.com repos will have @
+	// Pinned gitlab repos will have /-/
+	return (strings.Contains(repo, "@") || strings.Contains(repo, "/-/"))
+}
+
+func checkforUnpinnedRepos(validator *Validator) {
+	for i, component := range validator.typedZarfPackage.Components {
+		for j, repo := range component.Repos {
+			if !isPinnedRepo(repo) {
+				validator.addWarning(fmt.Sprintf(".components.[%d].repos.[%d]: Unpinned repository", i, j))
+			}
+		}
+	}
+}
+
+func checkForUnpinnedImages(validator *Validator) {
+	for i, component := range validator.typedZarfPackage.Components {
+		for j, image := range component.Images {
+			if !isPinnedImage(image) {
+				validator.addWarning(fmt.Sprintf(".components.[%d].images.[%d]: Unpinned image", i, j))
+			}
+		}
+	}
+}
+
+func checkForUnpinnedFiles(validator *Validator) {
+	for i, component := range validator.typedZarfPackage.Components {
+		for j, file := range component.Files {
+			if file.Shasum == "" {
```

Comment:
(already talked about this just a reminder to do isURL since we really only want to warn for remote sources)